receiver/prometheusreceiver/internal/transaction.go

```diff
@@ -27,27 +27,38 @@ import (
 )
 
 const (
-	targetMetricName = "target_info"
-	receiverName     = "otelcol/prometheusreceiver"
+	targetMetricName  = "target_info"
+	scopeMetricName   = "otel_scope_info"
+	scopeNameLabel    = "otel_scope_name"
+	scopeVersionLabel = "otel_scope_version"
+	receiverName      = "otelcol/prometheusreceiver"
 )
 
 type transaction struct {
-	isNew          bool
-	trimSuffixes   bool
-	ctx            context.Context
-	families       map[string]*metricFamily
-	mc             scrape.MetricMetadataStore
-	sink           consumer.Metrics
-	externalLabels labels.Labels
-	nodeResource   pcommon.Resource
-	logger         *zap.Logger
-	buildInfo      component.BuildInfo
-	metricAdjuster MetricsAdjuster
-	obsrecv        *obsreport.Receiver
+	isNew           bool
+	trimSuffixes    bool
+	ctx             context.Context
+	families        map[scopeID]map[string]*metricFamily
+	mc              scrape.MetricMetadataStore
+	sink            consumer.Metrics
+	externalLabels  labels.Labels
+	nodeResource    pcommon.Resource
+	scopeAttributes map[scopeID]pcommon.Map
+	logger          *zap.Logger
+	buildInfo       component.BuildInfo
+	metricAdjuster  MetricsAdjuster
+	obsrecv         *obsreport.Receiver
 	// Used as buffer to calculate series ref hash.
 	bufBytes []byte
 }
 
+var emptyScopeID scopeID
+
+type scopeID struct {
+	name    string
+	version string
```

Comment:
Do we want to include attributes in the ID, or we can always expect the scope name/version 1:1 correlation with its attributes?

Reply:
Scope attributes [are non-identifying](https://github.com/open-telemetry/opentelemetry-specification/blob/bc76c97525eef0f81550d939a3f87f44b182cef9/specification/glossary.md?plain=1#L155):

> The name or (name,version) pair uniquely identify the logical unit of the code that emits the telemetry.

Also, when we "join" otel_scope_info with other metrics, we can only join on otel_scope_name and otel_scope_version, as those are the only common labels.

That does leave the question of what to do if we get two otel_scope_info metrics with the same name and version.  Right now, we have last-writer wins behavior.  Appending is hard, because the attributes might conflict, and we would have to deal with merges.  First-writer wins seems no different than last-writer wins.

Reply:
Thanks for clarifying 👍